x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/TestCaseSupplier.java

```diff
@@ -357,6 +358,22 @@ public static List<TypedDataSupplier> getSuppliersForNumericType(DataType type,
         throw new IllegalArgumentException("bogus numeric type [" + type + "]");
     }
 
+    public static List<TypedDataSupplier> getSuppliersForNumericType(DataType type) {
```

Comment:
Should we just call this `getSuppliersForType()` and implement for everything? (Or leave it with an exception for unhandled cases until we need them)